homeassistant/components/rituals_perfume_genie/entity.py

```diff
@@ -24,7 +24,10 @@ class DiffuserEntity(CoordinatorEntity):
     """Representation of a diffuser entity."""
 
     def __init__(
-        self, diffuser: Diffuser, coordinator: CoordinatorEntity, entity_suffix: str
+        self,
```

Comment:
I can't see it. Should be here.
```python
class DiffuserEntity(CoordinatorEntity):
  coordinator: RitualsDataUpdateCoordinator
```

Reply:
I thought adding an annotation to the function parameter was enough.
I have added the field annotation in https://github.com/home-assistant/core/pull/49947/commits/5efca6b799144ada72bc28afb9ceac80f5999f4e.
Why is this necessary and why does mypy warn about this?

Reply:
> Why is this only necessary for the custom coordinator

This way mypy can know `self.coordinator` in entity has type `RitualsDataUpdateCoordinator` and not `DataUpdateCoordinator`.

> why does mypy not warn about this?

Because you didn't use `coordinator._device` outside `_async_update_data`. But if you add more fields there and start using them in entity, mypy won't able to find them. So it's just an extra annotation to let mypy know that `self.coordinator` is custom in entity.